packages/@aws-cdk/aws-stepfunctions-tasks/lib/dynamodb/shared-types.ts

```diff
@@ -108,82 +110,131 @@ export class DynamoProjectionExpression {
 }
 
 /**
- * Class to generate AttributeValue
+ * Map of string to AttributeValue
+ */
+export interface DynamoAttributeValueMap {
+  [key: string]: DynamoAttributeValue;
+}
+
+/**
+ * Represents the data for an attribute.
+ * Each attribute value is described as a name-value pair.
+ * The name is the data type, and the value is the data itself.
+ *
+ * @see https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_AttributeValue.html
  */
-export interface DynamoAttributeValue {
+export class DynamoAttributeValue {
   /**
    * Sets an attribute of type String. For example:  "S": "Hello"
-   *
-   * @default - None
+   * Strings may be literal values or as JsonPath
    */
-  readonly s?: string;
+  public static wrapString(s: string) {
+    return new DynamoAttributeValue({ S: s });
+  }
+
+  /**
+   * Sets a literal number. For example: 1234
+   * Numbers are sent across the network to DynamoDB as strings,
+   * to maximize compatibility across languages and libraries.
+   * However, DynamoDB treats them as number type attributes for mathematical operations.
+   */
+  public static wrapNumber(n: number) {
+    return new DynamoAttributeValue({ N: n.toString() });
+  }
 
   /**
    * Sets an attribute of type Number. For example:  "N": "123.45"
    * Numbers are sent across the network to DynamoDB as strings,
    * to maximize compatibility across languages and libraries.
    * However, DynamoDB treats them as number type attributes for mathematical operations.
    *
-   * @default - None
+   * Numbers may be expressed as literal strings or as JsonPath
    */
-  readonly n?: string;
+  public static wrapNumberFromString(n: string) {
+    return new DynamoAttributeValue({ N: n.toString() });
+  }
 
   /**
    * Sets an attribute of type Binary. For example:  "B": "dGhpcyB0ZXh0IGlzIGJhc2U2NC1lbmNvZGVk"
```

Comment:
Does the value need to be base64-encoded by the user? Will you do it for them?

Reply:
i should better document that, we did discuss this and whether we should do it on behalf of the user. the norm is for the user to supply a base 64-encoded string. 

do you feel otherwise?